### PR TITLE
fix: tags initially hidden, closing #11757

### PR DESCRIPTION
**Pull request by picodoth**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/tag/index.tsx [hunk 1]
```diff
@@ -130,7 +147,7 @@ class Tag extends React.Component<TagProps, TagState> {
       backgroundColor: (color && !isPresetColor) ? color : null,
       ...style,
     };
-    const tag = this.state.closed ? null : (
+    const tag = this.state.closed ? <div /> : (
```
Comment (picodoth) [inline components/tag/index.tsx]: <Wave /> requires element to be existed on componentDidMount

File: components/tag/demo/controlled.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { Tag, Button } from 'antd';
 
 class Demo extends React.Component {
   state = {
-    visible: true,
+    visible: false,
```
Comment (afc163) [inline components/tag/demo/controlled.md]: 默认不应该是 false。

File: components/tag/index.tsx [hunk 3]
```diff
@@ -37,14 +38,30 @@ class Tag extends React.Component<TagProps, TagState> {
...
-  static getDerivedStateFromProps(nextProps: TagProps) {
-    return  ('visible' in nextProps) ? { visible: nextProps.visible } : null;
+  static getDerivedStateFromProps(nextProps: TagProps, state: TagState) {
+    if ('visible' in nextProps) {
+      let newState: Partial<TagState> = {
+        visible: nextProps.visible,
+        mounted: true,
+      };
+
+      if (!state.mounted) {
```
Comment (picodoth) [inline components/tag/index.tsx]: only set closed state in component mounting phase

**Comment by picodoth** [comment]
顺带修复了 toggle animation
before:
before
 
after:
after

**Comment by yesmeck** [comment]
加个测试

**Comment by picodoth** [comment]
fix 如果没明显问题我就加测试了

**Comment by picodoth** [comment]
@yesmeck @afc163

**Comment by yesmeck** [comment]
看下 CI

**Pull request merged**